### ISSUE TITLE
Issue rendering partials

### ISSUE THREAD

**Issue by puffstream** [wombatsecurity/dust-loader-complete#15]
Hello, thanks for writing this loader. Just having some issues rendering partials. Getting the error cannot find module.

I have my webpack.config.js file set up. And a JavaScript file to handle rendering of a core template, and inside of this is a partial.

`
var dust = require('dustjs');
var template = require('../views/index.dust');
template('', callback);
function callback (err, result) {
	var rootClass = document.getElementsByClassName('root')[0];
	rootClass.innerHTML = result;
}
`

And in my template is this...
< div >
 {>"somePartial"/}
< / div >

Any ideas? Appreciate any help. Thanks.

**Comment by jrrbru** [comment]
Hi James, thanks for taking a look at the project.

My guess is that the issue is you haven't set an appropriate `root` in the loader options: https://github.com/wombatsecurity/dust-loader-complete#root. The loader turns the template's path into a template name. By setting the `root`, we remove everything between the webpack root directory and the specified `root` directory from the name.

Let me know if that helps (or not).

**Comment by puffstream** [comment]
Okay, makes sense. I've started going down another track for dustjs implementation with Webpack, but can take another look in future. Thanks!!

**Comment by farshadahmadi** [comment]
Hi,

I have (had ?) the same problem. When I use a partial, I encounter an error saying the "Module not found". I created a minimal github [project ](https://github.com/farshadahmadi/test-dust-loader-complete) that throw the error. Just clone it, do the `npm install`, and open the http://localhost:8080/index.html. 

The following is my **webpack.config.js** file:

```
const path = require('path');

module.exports = {
  context: path.join(__dirname,"src"),
  entry: {
    app: "./app.js"
  },
  output: {
    path: path.join(__dirname,"dist"),
    filename: '[name].bundle.js',
    publicPath: 'http://localhost:8080/dist/'
  },
  resolve: {
    modules: [path.join(__dirname, 'node_modules')],
    alias: {
      'dustjs' : 'dustjs-linkedin'
    }
  },
  module: {
    rules: [
      {
        test: /\.dust$/,
        use : [
          {
            loader: "dust-loader-complete",
            options:{
              root: path.join(__dirname, "src","templates"),
              verbose: true,
              dustAlias: "dustjs"
            }
          }
        ]
      }
    ]
  }
};
```

Then I have the following **hello-world.dust** file:

```
<p>Hello {name}!</p>
{> "partial-template" /}
```
and **partial-template.dust** contains a simple html element:

`<p>Hello from partial</p>`

The **app.js** file contains:

```
var helloWorldTemplate = require("./templates/hello-world.dust");

var dust = require("dustjs");
var $ = require("jquery");

 // When th page is loaded
$(function() {

	dust.render(helloWorldTemplate, {name: "world"}, function(err, html){
	    if(err){
	      console.err(err);
	    } else {
	      $('#greeting').html(html);
	    }
	});

});
```

and finally the `index.html` file contains:

```
<!doctype html>
<body>
  <script src="http://localhost:8080/dist/app.bundle.js"></script>
  <div id="greeting"></div>
</body>
</html>
```

Starting webpack-dev-server throws the following error;

```
ERROR in ./src/templates/hello-world.dust
Module not found: Error: Can't resolve 'partial-template' in '/Users/farshad/Documents/webpack/src/templates'
 @ ./src/templates/hello-world.dust 1:45-72
 @ ./src/app.js
 @ multi (webpack)-dev-server/client?http://localhost:8080 ./app.js
```

Then I went through the code (**index.js** file) of dust-loader-complete. I found the problem in **addDustDependency** function.
```

1. // Add a dust dependency to the dependency array, returning the normalized path/name
2. function addDustDependency( require_path, source_path, options, deps ) {
3. var name = determinePartialName( require_path, source_path, options );
4. deps.push( "var partial" + deps.length + " = require('" + name + "');" );
5. return name;
6. }

```

The name variable in line 4 cannot be resolved. Then I change that line to contain the absolute path together with **.dust** extension.

4.   //deps.push( "var partial" + deps.length + " = require('" + path.resolve(options.root, name + ".dust") + "');" );

and it worked! I do not know if there is a better way than this workaround!

@jrrbru Do you have any idea?

Thanks

**Comment by jrrbru** [comment]
Hi @farshadahmadi I think I found the issue. I was able to get your project to work by making the following changes to the `resolve` node in `webpack.config.js`:

    modules: [path.join(__dirname, 'node_modules'), path.join(__dirname, 'src/templates')],
    extensions: ['.js','.dust'],

This is necessary because a) the partials are being converted into "absolute" requires based on the `root` option, and webpack needs to know where to look for them, and b) the requires by default omit the `.dust` extensions.

I think I may fix these issues in a future update.
